server/server.go

```diff
@@ -241,6 +241,11 @@ func (u *UpdateServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// New versions don't support Windows XP any more.
+	if params.OS == "windows" && params.Arch == "386" {
+		u.closeWithStatus(w, http.StatusNoContent)
+		return
+	}
```

Comment:
I don't actually think this works unfortunately @joesis because we compile all of our windows binaries to be 386:

https://github.com/getlantern/lantern-build/blob/devel/Makefile#L234

Reply:
We might have to add an argument that tells us if the actual OS is XP and shoot that in to the autoupdate server and when that's widespread enough (auto-update to that version), then we can start ignoring it.

Reply:
Ahhhh you are right, it's a nuclear bomb.

You just reminded me that we can do this on the client side - not running auto-update module at all on XP or OSX 10, and even better, we can pop up notification to the users.

Reply:
Or on the safer side, we can plan to stop supporting XP/OSX 10 at a future version, say, 5.4.0, and embed the logic of not auto-updating to 5.4.0 or above in the clients **now**. It would let basically all clients know to NOT update before actually dropping the support.